kong/plugins/tcp-log/handler.lua

```diff
@@ -43,6 +43,29 @@ function TcpLogHandler:new()
   TcpLogHandler.super.new(self, "tcp-log")
 end
 
+function TcpLogHandler:access(conf)
+  TcpLogHandler.super.access(self)  
+
+  if conf.log_body then
+    ngx.req.read_body()
+    ngx.ctx.req_resp_bodies = {
```

Comment:
Instead of a single `ctx` variable containing two fields, I think it would make more sense to have two separate values:

```
ngx.ctx.reqest_body = ngx.req.get_body_data()
ngx.ctx.response_body = ""
```